src/app/(app)/course/[courseId]/(hooks)/useCourse.tsx

```diff
@@ -40,6 +41,10 @@ const useCourseProvider = (): CourseContextType => {
 
 export const CourseProvider: React.FC<PropsWithChildren> = ({ children }) => {
   const courseContext = useCourseProvider()
+  if (!courseContext.courseId) {
+    return Custom404({ errorMessage: "Course not found" })
```

Comment:
It feels weird to me that we are not calling this function in the `<Component />` way